app/domain/authentication/authn_k8s/inject_client_cert.rb

```diff
@@ -27,9 +27,24 @@ def call
 
       private
 
+      # In the old version of the authn-client we assumed that the host is under the "apps" policy branch.
+      # Now we send the host-id in 2 parts:
+      #   suffix - the host id
+      #   prefix - the policy id
+      # We update the CSR's common_name to have the full host-id. This way, the validation
+      # that happens in the "authenticate" request will work, as the signed certificate
+      # contains the full host-id.
+      def update_csr_common_name
+        prefix = @host_id_prefix.nil? || @host_id_prefix.empty? ? apps_host_id_prefix : @host_id_prefix
+        full_host_name = prefix + "." + smart_csr.common_name
+
+        Rails.logger.debug(Log::SetCommonName.new(full_host_name))
```

Comment:
`Rails.logger` should be a dependency.

Reply:
Correct, it's already there. changing this to use it.